app/src/main/java/com/calendar/Activity/CountDays.java

```diff
@@ -0,0 +1,153 @@
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.util.Pair;
+
+import com.calendar.R;
+import com.google.android.material.datepicker.MaterialDatePicker;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Objects;
+
+public class CountDays extends AppCompatActivity {
+
+    private static final DatabaseReference databaseReference = FirebaseDatabase.getInstance()
+            .getReference("Calendar");
+
+    private Button dataRangeButton;
+    private TextView dateRangeText;
+    private TextView numberOfFreeDaysText;
+    private TextView numberOfWorkingDaysText;
+    private MaterialDatePicker<Pair<Long, Long>> materialDatePicker;
+    private int numberOfFreeDays = 0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.count_days);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        setGuiComponents();
+        setMaterialDatePicker();
+    }
+
+    private void setGuiComponents() {
+        dataRangeButton = findViewById(R.id.dateRangeButton);
+        dateRangeText = findViewById(R.id.dateRangeText);
+        numberOfFreeDaysText = findViewById(R.id.numberOfFreeDays);
+        numberOfWorkingDaysText = findViewById(R.id.numberOfWorkingDays);
+    }
+
+    private void setMaterialDatePicker() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.DAY_OF_MONTH, 1);
+        long monthStart = calendar.getTimeInMillis();
+        calendar.set(Calendar.DAY_OF_MONTH, calendar.getActualMaximum(Calendar.DAY_OF_MONTH));
+        long monthEnd = calendar.getTimeInMillis();
+
+        materialDatePicker = MaterialDatePicker.Builder
+                .dateRangePicker()
+                .setSelection(new Pair<>(monthStart, monthEnd))
+                .build();
+        setMaterialDatePickerListener();
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private void setMaterialDatePickerListener() {
+        materialDatePicker.addOnPositiveButtonClickListener(selection -> {
+            dateRangeText.setText(materialDatePicker.getHeaderText());
+            Date startDate = new Date(selection.first);
+            Date endDate = new Date(selection.second);
+            LocalDate startLocalDate = dateToLocalDate(startDate);
+            LocalDate endLocalDate = dateToLocalDate(endDate);
+            countDays(startLocalDate, endLocalDate);
+        });
+    }
+
+    public void selectDateRange(View view) {
```

Comment:
I think this method is not neceserry.

Reply:
This method is attached to button and is responsible for showing and allowing user to change selected date range. For what I know it is necessary to allow this operation.